### PR TITLE
Emit llvm.is.fpclass instead of llvm.amdgcn.class

### PR DESCRIPTION
**Pull request by jayfoad**
This enables more optimizations in LLVM, like combining multiple class tests or comparisons into a single llvm.is.fpclass operation.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit eba505f61efb8af5e9067a66bbbc7c2ca7fa0c5b
<details><summary>CTS tests (Failed: 0/208297)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 40199/69439 (57.9%)</li>
<li>Failed: 0/69439 (0.0%)</li>
<li>Not Supported: 29240/69439 (42.1%)</li><li>Warnings: 0/69439 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 40225/69417 (57.9%)</li>
<li>Failed: 0/69417 (0.0%)</li>
<li>Not Supported: 29192/69417 (42.1%)</li><li>Warnings: 0/69417 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 36765/69441 (52.9%)</li>
<li>Failed: 0/69441 (0.0%)</li>
<li>Not Supported: 32676/69441 (47.1%)</li><li>Warnings: 0/69441 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by jayfoad** [comment]
https://reviews.llvm.org/D148803 caused another prolem that should be fixed by https://reviews.llvm.org/D148905.

**Review by nhaehnle — COMMENTED** [review]
Small nit inline, but apart from that LGTM assuming all the LLVM issues are worked through ;)

**Review by nhaehnle — APPROVED** [review]
Thanks

**Comment by amdvlk-admin** [comment]
### Test summary for commit f0011d98e19cf96ef6239f82608dfda13176cb79
<details><summary>CTS tests (Failed: 0/215998)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 40685/72006 (56.5%)</li>
<li>Failed: 0/72006 (0.0%)</li>
<li>Not Supported: 31321/72006 (43.5%)</li><li>Warnings: 0/72006 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 40713/71984 (56.6%)</li>
<li>Failed: 0/71984 (0.0%)</li>
<li>Not Supported: 31271/71984 (43.4%)</li><li>Warnings: 0/71984 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 36765/72008 (51.1%)</li>
<li>Failed: 0/72008 (0.0%)</li>
<li>Not Supported: 35243/72008 (48.9%)</li><li>Warnings: 0/72008 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by jayfoad** [comment]
The LLVM changes have percolated through now so all checks are passing. Just waiting for re-approval.

**Review by trenouf — APPROVED** [review]
LGTM

**Pull request merged**
